File: api_automations/domain_checker/python/domain_avail_checker_solo.py
import argparse
import time
import os

import requests

# set up cli domain argument
parser = argparse.ArgumentParser(description="Check domain for availability")
parser.add_argument("domain", type=str, help="Domain name to be checked")
args = parser.parse_args()

# godaddy API credentials for authoriztation
api_key = "3mM44UahMcshjE_R1iX6AkK5L14Q9CYPNfyw6"
api_secret = "WmHxMWoB2BvBBuyMh7qCuc"
req_headers = {"Authorization": f"sso-key {api_key}:{api_secret}"}


# request the development instance of godaddy for the domain
def get_req_url(check_domain):
    return f"https://api.ote-godaddy.com/v1/domains/available?domain={check_domain}"


def check_domain_available(check_domain):
    print(f"Checking availability of domain {check_domain}")
    req_url = get_req_url(check_domain)
    req = requests.get(req_url, headers=req_headers)

    # if the request was unsuccessful, notify and return
    if req.status_code != 200:
        print(
            f"Could not get availability state of domain {check_domain} - Status Code {req.response_code}"
        )
        return

    # check if the domain is available
    response = req.json()
    if response["available"] == True:
        print(f"Domain {check_domain} is available for purchase")

    else:
        print(
            f"{time.strftime('%Y-%m-%d %H:%M')} - Domain {check_domain} is not available for purchase."
        )


check_domain_available(args.domain)

File: api_automations/domain_checker/javascript/domain_avail_checker_solo.js
// npm install argparse axios
const axios = require('axios').default;
const { ArgumentParser } = require('argparse');

const parser = new ArgumentParser({ description: 'Check domain for availability' });
parser.addArgument('domain', {
  type: 'string',
  help: 'Domain name to be checked',
});

const { domain } = parser.parseArgs();
const apiKey = 'ENTER API KEY HERE';
const apiSecret = 'ENTER API SECRET HERE';
const headers = { Authorization: `sso-key ${apiKey}:${apiSecret}` };

const getUrl = domain => `https://api.ote-godaddy.com/v1/domains/available?domain=${domain}`;

async function checkDomainAvailability(domain) {
  console.log(`Checking availability of domain ${domain}`);
  const res = await axios.get(getUrl(domain), { headers });

  if (res.status != 200) {
    console.log(`Error getting state of Domain ${domain}`);
    return;
  }

  const available = res.data.available;
  if (!available) {
    console.log(`Domain ${domain} is not available.`);
    return;
  }

  console.log(`Domain ${domain} is available for purchase`);
}

checkDomainAvailability(domain);